framework/include/problems/FEProblemBase.h

```diff
@@ -1861,6 +1863,8 @@ class FEProblemBase : public SubProblem, public Restartable
   /// requested an AD material property or whether any suppier has declared an AD material property
   bool _using_ad_mat_props;
 
+  bool _hypre_matrix;
```

Comment:
I would prefer _reuse_hypre_matrix for a Boolean. As it stands, it's a little confusing and you might think it's a pointer to the matrix.